File: csharp/resistor-color/ResistorColor.cs
﻿using System;

public static class ResistorColor
{
    public static int ColorCode(string color) => Array.IndexOf(Colors(), color);

    public static string[] Colors()
    {
        string[] colorList = { "black", "brown", "red", "orange", "yellow", 
                                "green", "blue", "violet", "grey", "white" };
        return colorList;
    }
}

File: kotlin/resistor-color/src/main/kotlin/ResistorColor.kt
object ResistorColor {

    private val codes: List<String> = listOf("black","brown","red","orange","yellow","green","blue","violet","grey","white")

    fun colorCode(input: String): Int {
        return codes.indexOf(input)
    }

    fun colors(): List<String> {
        return codes
    }

}
